conwayste/src/uimanager.rs

```diff
@@ -0,0 +1,73 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of conwayste.
+ *
+ *  conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with conwayste.  If not, see
+ *  <http://www.gnu.org/licenses/>. */
+
+use crate::Screen;
+use crate::uilayout::UILayout;
+use crate::ui::{
+    Chatbox,
+    Layer,
+    TextField,
+    WidgetID,
+};
+
+pub struct LayoutManager;
+
+impl LayoutManager {
```

Comment:
FWIW I have a long-term goal to make these functions generically reusable, but I decided to leave it for another time. This PR is already aging.